In [1]:
%load_ext autoreload
%autoreload 2

from helpers import *

from IPython.display import clear_output

import bz2
import json

import pandas as pd

import numpy as np

from pprint import *

from perform_general_analysis import *

import matplotlib.pyplot as plt

## Raw Data Exploration

We compute some simple stats on the raw dataset(s). Since, they are so large
they need to be loaded in chunks and various starts are aggregated after 
each chunk's computation. This should give a better understand of the
data we are working with. The code used for the chunk extraction and stats calculation is contained in perform_general_analysis.py.

General interesting stats for all the quote banks dataset from 2015 to 2020

In [ ]:
all_paths = [QUOTES_2015_PATH,QUOTES_2016_PATH,QUOTES_2017_PATH,QUOTES_2018_PATH,QUOTES_2019_PATH,QUOTES_2020_PATH]
all_stats = []
for path in all_paths:
    stat_per_year = build_stat_dics(path)
    all_stats.append(stat_per_year)

In [7]:
all_stats

[{'year': '2015',
  'count': 20874338,
  'count_no_speaker': 7016300,
  'quotation': {'chars': {'min_length': 5,
    'max_length': 4525,
    'sum_length': 2698946982,
    'avg_length': 129.29},
   'words': {'min_length': 1,
    'max_length': 471,
    'sum_length': 490585455,
    'avg_length': 23.5}}},
 {'year': '2016',
  'count': 13862129,
  'count_no_speaker': 4755415,
  'quotation': {'chars': {'min_length': 7,
    'max_length': 3064,
    'sum_length': 1789967993,
    'avg_length': 129.13},
   'words': {'min_length': 1,
    'max_length': 469,
    'sum_length': 325730694,
    'avg_length': 23.5}}},
 {'year': '2017',
  'count': 26611588,
  'count_no_speaker': 9278290,
  'quotation': {'chars': {'min_length': 6,
    'max_length': 2995,
    'sum_length': 3506275835,
    'avg_length': 131.76},
   'words': {'min_length': 1,
    'max_length': 472,
    'sum_length': 636423153,
    'avg_length': 23.92}}},
 {'year': '2018',
  'count': 27228451,
  'count_no_speaker': 9558032,
  'quotation': {'cha

Now, lets aggregate all the years to get a full statistics over all the entire dataset

In [8]:
stats_dataset = build_total_dict(all_stats)
stats_dataset

{'count': 115584257,
 'count_no_speaker': 39988889,
 'quotation': {'words': {'min_length': 1,
   'max_length': 483,
   'sum_length': 2781157961,
   'avg_length': 24.061736720771584},
  'chars': {'min_length': 5,
   'max_length': 10449,
   'sum_length': 15349595389,
   'avg_length': 132.8000524240944}}}

In [9]:
#percentage of no speakers quotes along all dataset
percentage_no_speakers = stats_dataset["count_no_speaker"]/stats_dataset["count"]
percentage_no_speakers

0.34597176153496406

We retrieved the following infos:
- the quotebank dataset from 2015 to 2020 contain around 115milions of entries
- 34.5% of the quotes of the dataset have no definitive speaker attributed
- the number of words per quote span from a minium of 1 to a maximum of 483 words, where the average number is 24 words
- the number of characters per quote span from a minium of 5 to a maximum of 10449 characters, where the average number is 132 characters

Below are the distributions of number of characters and words per quote for the entire dataset